pandas/_libs/algos.pyx

```diff
@@ -1158,34 +1157,14 @@ def rank_2d(
 
     n, k = (<object>values).shape
     ranks = np.empty((n, k), dtype='f8')
-    # For compatibility when calling rank_1d
-    labels = np.zeros(k, dtype=np.int64)
 
-    if rank_t is object:
-        try:
-            _as = values.argsort(1)
-        except TypeError:
-            values = in_arr
-            for i in range(len(values)):
-                ranks[i] = rank_1d(
-                    in_arr[i],
-                    labels=labels,
-                    ties_method=ties_method,
-                    ascending=ascending,
-                    pct=pct
-                )
-            if axis == 0:
-                return ranks.T
```

Comment:
This whole try/except is not necessary because the except clause will always raise a `TypeError` anyway since if `values.argsort(1)` raises a `TypeError`, at least one column in `values` must not be sortable, so that same column will cause another `TypeError` to be raised when trying to fallback to `rank_1d`